teps/0094-configuring-resources-at-runtime.md

```diff
@@ -52,12 +61,362 @@ While this may be a valuable feature, it should be considered in a separate prop
+need to override additional `Container` fields in TaskRuns.
+
+However, at the moment, there is no clear use case for such features,
+and the ability to override `Step` and `Sidecar` fields
+other than `Resources` is a non-goal of this proposal.
+
+```go
+import corev1 "k8s.io/api/core/v1"
+
+// Container configuration that will override configuration specified in a Task or PipelineTask.
+type ContainerOverride struct {
+   Resources corev1.ResourceRequirements
+}
+
+type TaskRunSpec struct {
+   ...
+   // A map of Step names to ContainerOverrides to apply to that Step.
+   // A single entry with key "*" will apply the ContainerOverride to all Steps.
+   // Any field present in both a Step and its ContainerOverride
+   // will use the value from the ContainerOverride.
+   StepOverrides map[string]ContainerOverride
+
+   // A map of Sidecar names to ContainerOverrides to apply to that Sidecar.
+   // A single entry with key "*" will apply the ContainerOverride to all Sidecars.
+   // Any field present in both a Sidecar and its ContainerOverride
+   // will use the value from the ContainerOverride.
+   SidecarOverrides map[string]ContainerOverride
+}
+
+type PipelineTaskRunSpec struct {
+  ...
+   // A map of Step names to ContainerOverrides to apply to that Step.
+   // A single entry with key "*" will apply the ContainerOverride to all Steps.
+   // Any field present in both a Step and its ContainerOverride
+   // will use the value from the ContainerOverride.
+   StepOverrides map[string]ContainerOverride
+
+   // A map of Sidecar names to ContainerOverrides to apply to that Sidecar.
+   // A single entry with key "*" will apply the ContainerOverride to all Sidecars.
+   // Any field present in both a Sidecar and its ContainerOverride
+   // will use the value from the ContainerOverride.
+   SidecarOverrides map[string]ContainerOverride
+}
+```
+
+### Risks and Mitigations
+
+This proposal may increase confusion for users. If implemented, there will be three ways
+to specify `Step` resource requirements: in `Task.Step.Resources`, `Task.StepTemplate.Resources`,
+and `TaskRun.StepResources`. It may not be obvious which takes precedence.
+This risk can be mitigated via documentation and good naming.
+
+### User Experience
+
+The current workaround for lack of this feature is to write a new `Task`
+for each set of resource constraints, as described in
+[this comment](https://github.com/tektoncd/pipeline/issues/4080#issuecomment-884958486)
+from a buildah user.
+
+This proposal moves environment-specific configuration into `TaskRun` definitions,
+allowing `Task` definitions to be reused.
+We may want to add an option to the CLI to specify resource requirements when starting
+a `TaskRun` via `tkn task start`, but this is not necessary for the initial implementation.
+
+## Test Plan
+
+E2E tests should include `Task`s with and without resource requirements specified
```

Comment:
FWIW, you could probably just rely on unit tests, since you'll know if the ResourceRequest evaluation is right based on the Pod that's produced (that said, +1 for adding an example to reference in docs).